zynqmp/optee-os/optee-os.bb

```diff
@@ -0,0 +1,67 @@
+OPTEE_VERSION ??= "latest"
+SRCREV ??= "${AUTOREV}"
+BRANCH ??= "master"
+
+DESCRIPTION = "OP-TEE OS"
+
+# Define as closed license to prevent MD5 checksum verification since
+# LICENSE file changed around 3.5.0 making this recipe less flexible.
+# LICENSE = "BSD"
+# ante 3.5.0: LIC_FILES_CHKSUM = "file://LICENSE;md5=69663ab153298557a59c67a60a743e5b"
+# post 3.5.0: LIC_FILES_CHKSUM = "file://LICENSE;md5=c1f21c4f72f372ef38a5a4aee55ec173"
+LICENSE = "CLOSED"
+
+inherit deploy
+
+PROVIDES = "virtual/optee-os"
+DEPENDS += "u-boot-mkimage-native"
+
+S = "${WORKDIR}/git"
+PV = "${OPTEE_VERSION}+git${SRCPV}"
+
+REPO ??= "git://github.com/OP-TEE/optee_os.git;protocol=git"
+SRC_URI = "${REPO};branch=${BRANCH}"
+
+OPTEE_BASE_NAME ?= "${PN}-${PKGE}-${PKGV}-${PKGR}-${DATETIME}"
+OPTEE_BASE_NAME[vardepsexclude] = "DATETIME"
+
+# requires CROSS_COMPILE set by hand as there is no configure script
+export CROSS_COMPILE="${TARGET_PREFIX}"
+
+# Let the Makefile handle setting up the CFLAGS and LDFLAGS as it is a standalone application
+CFLAGS[unexport] = "1"
+LDFLAGS[unexport] = "1"
+AS[unexport] = "1"
+LD[unexport] = "1"
+
+TA_DEV_KIT_DIR = "${TMPDIR}/deploy/images/${MACHINE}/optee/export-ta_arm64"
+OUTPUT_DIR = "${S}/out/arm-plat-zynqmp"
+
+EXTRA_OEMAKE_append = " comp-cflagscore=--sysroot=${STAGING_DIR_HOST}"
+EXTRA_OEMAKE_append = " CROSS_COMPILE=${CROSS_COMPILE}"
+EXTRA_OEMAKE_append = " CROSS_COMPILE_core=${CROSS_COMPILE}"
+EXTRA_OEMAKE_append = " CROSS_COMPILE_ta_arm64=${CROSS_COMPILE}"
+EXTRA_OEMAKE_append = " PLATFORM=zynqmp"
+EXTRA_OEMAKE_append = " CFG_ARM64_core=y"
+EXTRA_OEMAKE_append = " CFG_ARM32_core=n"
+EXTRA_OEMAKE_append = " CFG_USER_TA_TARGETS=ta_arm64"
+EXTRA_OEMAKE_append = " CFG_TEE_BENCHMARK=n"
+EXTRA_OEMAKE_append = " CFG_TEE_CORE_LOG_LEVEL=1"
+EXTRA_OEMAKE_append = " CFG_TEE_CORE_DEBUG=n"
+EXTRA_OEMAKE_append = " DEBUG=0"
```

Comment:
Maybe relate these debug info to `DEBUG` state, as in tf-a recipe.

Reply:
In TF-A `DEBUG` is useful because it can be set trough petalinux GUI which is not the case here. This why I removed line of code related to debug state. However I can add a variable to allow debug state overriding trough a bbappend file.